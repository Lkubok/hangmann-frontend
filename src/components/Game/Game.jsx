import React, { Component } from "react";
import * as selectors from "../../reducers/selectors";
import { connect } from "react-redux";
import StartGame from "./StartGame/index";
import GameInProgress from "./GameInProgress/GameInProgress";
export class Game extends Component {
  render() {
    return this.props.gameId ? <GameInProgress /> : <StartGame />;
  }
}
const mapStateToProps = state => ({
  gameId: selectors.getGameId(state)
});

export default connect(mapStateToProps)(Game);
