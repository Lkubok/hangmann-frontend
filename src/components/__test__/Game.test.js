import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Game } from "../Game/Game";
Enzyme.configure({ adapter: new Adapter() });

describe("Game component", () => {
  beforeEach(() => {});
  it("Should match snapshot", () => {
    const wrapper = shallow(<Game />);
    expect(wrapper).toMatchSnapshot();
  });
});
