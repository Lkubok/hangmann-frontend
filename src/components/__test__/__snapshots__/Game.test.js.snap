// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game component Should match snapshot 1`] = `<Connect(WithFormik(StartGame)) />`;
